### ISSUE TITLE
Log stopping problem with no response.

### ISSUE THREAD

**Issue by celestialxevermore** [ArrowLuo/CLIP4Clip#79]
Dear Author, 

I hope almost done with running your framework, and It should be last question from me, 

I saw some successful step to run, but I found that any log is written in log.txt. 
(By the way, I installed conda install pytorch==1.8.0 torchvision==0.9.0 torchaudio==0.8.0 cudatoolkit=11.1 -c pytorch -c conda-forge, it worked for me)

If you can figure out what is wrong on my circumstance, I hope your help :) 
Thx in advance. 

(CLIP4Clip) key2317@super:~/CLIP4Clip$ DATA_PATH=/home/key2317/CLIP4Clip/msvd_data
(CLIP4Clip) key2317@super:~/CLIP4Clip$ CUDA_VISIBLE_DEVICES=3,4,5,6 python -m torch.distributed.launch --nproc_per_node=4 \
> main_task_retrieval.py --do_train --num_thread_reader=2 \
> --epochs=5 --batch_size=128 --n_display=50 \
> --data_path ${DATA_PATH} \
> --features_path ${DATA_PATH}/MSVD_Videos \
> --output_dir ckpts/ckpt_msvd_retrieval_looseType \
> --init_model ckpts/ckpt_msvd_retrieval_looseType/pytorch_model.bin.2 \
> --lr 1e-4 --max_words 32 --max_frames 12 --batch_size_val 16 \
> --datatype msvd \
> --feature_framerate 1 --coef_lr 1e-3 \
> --freeze_layer_num 0 --slice_framepos 2 \
> --loose_type --linear_patch 2d --sim_header meanP \
> --pretrained_clip_name ViT-B/32 
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
INFO:logger:device: cuda:1 n_gpu: 4
INFO:logger:device: cuda:3 n_gpu: 4
INFO:logger:device: cuda:2 n_gpu: 4
INFO:logger:Effective parameters:
INFO:logger:  <<< batch_size: 128
INFO:logger:  <<< batch_size_val: 16
INFO:logger:  <<< cache_dir: 
INFO:logger:  <<< coef_lr: 0.001
INFO:logger:  <<< cross_model: cross-base
INFO:logger:  <<< cross_num_hidden_layers: 4
INFO:logger:  <<< data_path: /home/key2317/CLIP4Clip/msvd_data
INFO:logger:  <<< datatype: msvd
INFO:logger:  <<< do_eval: False
INFO:logger:  <<< do_lower_case: False
INFO:logger:  <<< do_pretrain: False
INFO:logger:  <<< do_train: True
INFO:logger:  <<< epochs: 5
INFO:logger:  <<< eval_frame_order: 0
INFO:logger:  <<< expand_msrvtt_sentences: False
INFO:logger:  <<< feature_framerate: 1
INFO:logger:  <<< features_path: /home/key2317/CLIP4Clip/msvd_data/MSVD_Videos
INFO:logger:  <<< fp16: False
INFO:logger:  <<< fp16_opt_level: O1
INFO:logger:  <<< freeze_layer_num: 0
INFO:logger:  <<< gradient_accumulation_steps: 1
INFO:logger:  <<< hard_negative_rate: 0.5
INFO:logger:  <<< init_model: ckpts/ckpt_msvd_retrieval_looseType/pytorch_model.bin.2
INFO:logger:  <<< linear_patch: 2d
INFO:logger:  <<< local_rank: 0
INFO:logger:  <<< loose_type: True
INFO:logger:  <<< lr: 0.0001
INFO:logger:  <<< lr_decay: 0.9
INFO:logger:  <<< margin: 0.1
INFO:logger:  <<< max_frames: 12
INFO:logger:  <<< max_words: 32
INFO:logger:  <<< n_display: 50
INFO:logger:  <<< n_gpu: 1
INFO:logger:  <<< n_pair: 1
INFO:logger:  <<< negative_weighting: 1
INFO:logger:  <<< num_thread_reader: 2
INFO:logger:  <<< output_dir: ckpts/ckpt_msvd_retrieval_looseType
INFO:logger:  <<< pretrained_clip_name: ViT-B/32
INFO:logger:  <<< rank: 0
INFO:logger:  <<< resume_model: None
INFO:logger:  <<< sampled_use_mil: False
INFO:logger:  <<< seed: 42
INFO:logger:  <<< sim_header: meanP
INFO:logger:  <<< slice_framepos: 2
INFO:logger:  <<< task_type: retrieval
INFO:logger:  <<< text_num_hidden_layers: 12
INFO:logger:  <<< train_csv: data/.train.csv
INFO:logger:  <<< train_frame_order: 0
INFO:logger:  <<< use_mil: False
INFO:logger:  <<< val_csv: data/.val.csv
INFO:logger:  <<< video_dim: 1024
INFO:logger:  <<< visual_num_hidden_layers: 12
INFO:logger:  <<< warmup_proportion: 0.1
INFO:logger:  <<< world_size: 4
INFO:logger:device: cuda:0 n_gpu: 4
WARNING:modules.modeling:Stage-One:True, Stage-Two:False
WARNING:modules.modeling:Test retrieval by loose type.
WARNING:modules.modeling:        embed_dim: 512
WARNING:modules.modeling:        image_resolution: 224
WARNING:modules.modeling:        vision_layers: 12
WARNING:modules.modeling:        vision_width: 768
WARNING:modules.modeling:        vision_patch_size: 32
WARNING:modules.modeling:        context_length: 77
WARNING:modules.modeling:        vocab_size: 49408
WARNING:modules.modeling:        transformer_width: 512
WARNING:modules.modeling:        transformer_heads: 8
WARNING:modules.modeling:        transformer_layers: 12
WARNING:modules.modeling:                linear_patch: 2d
WARNING:modules.modeling:        cut_top_layer: 0
WARNING:modules.modeling:        sim_header: meanP
For test, sentence number: 27763
For test, video number: 670
Video number: 670
Total Paire: 27763
/home/key2317/anaconda3/envs/CLIP4Clip/lib/python3.8/site-packages/torchvision/transforms/transforms.py:257: UserWarning: Argument interpolation should be of type InterpolationMode instead of int. Please, use InterpolationMode enum.
  warnings.warn(
For val, sentence number: 4290
For val, video number: 100
Video number: 100
Total Paire: 4290
Video number: 1200
Total Paire: 48774
For test, sentence number: 27763
For test, video number: 670
Video number: 670
Total Paire: 27763
/home/key2317/anaconda3/envs/CLIP4Clip/lib/python3.8/site-packages/torchvision/transforms/transforms.py:257: UserWarning: Argument interpolation should be of type InterpolationMode instead of int. Please, use InterpolationMode enum.
  warnings.warn(
For val, sentence number: 4290
For val, video number: 100
Video number: 100
Total Paire: 4290
Video number: 1200
Total Paire: 48774
For test, sentence number: 27763
For test, video number: 670
Video number: 670
Total Paire: 27763
/home/key2317/anaconda3/envs/CLIP4Clip/lib/python3.8/site-packages/torchvision/transforms/transforms.py:257: UserWarning: Argument interpolation should be of type InterpolationMode instead of int. Please, use InterpolationMode enum.
  warnings.warn(
For test, sentence number: 27763
For test, video number: 670
Video number: 670
Total Paire: 27763
/home/key2317/anaconda3/envs/CLIP4Clip/lib/python3.8/site-packages/torchvision/transforms/transforms.py:257: UserWarning: Argument interpolation should be of type InterpolationMode instead of int. Please, use InterpolationMode enum.
  warnings.warn(
For val, sentence number: 4290
For val, video number: 100
Video number: 100
Total Paire: 4290
For val, sentence number: 4290
For val, video number: 100
Video number: 100
Total Paire: 4290
INFO:logger:***** Running test *****
INFO:logger:  Num examples = 27763
INFO:logger:  Batch size = 16
INFO:logger:  Num steps = 1736
INFO:logger:***** Running val *****
INFO:logger:  Stopping Here..Num examples = 4290
Video number: 1200
Total Paire: 48774
Video number: 1200
Total Paire: 48774

nothing after it. 

Also, my log is   

2022-06-14 15:06:42,968:INFO: Effective parameters:
2022-06-14 15:06:42,968:INFO:   <<< batch_size: 128
2022-06-14 15:06:42,968:INFO:   <<< batch_size_val: 16
2022-06-14 15:06:42,968:INFO:   <<< cache_dir: 
2022-06-14 15:06:42,968:INFO:   <<< coef_lr: 0.001
2022-06-14 15:06:42,968:INFO:   <<< cross_model: cross-base
2022-06-14 15:06:42,968:INFO:   <<< cross_num_hidden_layers: 4
2022-06-14 15:06:42,969:INFO:   <<< data_path: /home/key2317/CLIP4Clip/msvd_data
2022-06-14 15:06:42,969:INFO:   <<< datatype: msvd
2022-06-14 15:06:42,969:INFO:   <<< do_eval: False
2022-06-14 15:06:42,969:INFO:   <<< do_lower_case: False
2022-06-14 15:06:42,969:INFO:   <<< do_pretrain: False
2022-06-14 15:06:42,969:INFO:   <<< do_train: True
2022-06-14 15:06:42,969:INFO:   <<< epochs: 5
2022-06-14 15:06:42,969:INFO:   <<< eval_frame_order: 0
2022-06-14 15:06:42,969:INFO:   <<< expand_msrvtt_sentences: False
2022-06-14 15:06:42,969:INFO:   <<< feature_framerate: 1
2022-06-14 15:06:42,969:INFO:   <<< features_path: /home/key2317/CLIP4Clip/msvd_data/MSVD_Videos
2022-06-14 15:06:42,969:INFO:   <<< fp16: False
2022-06-14 15:06:42,970:INFO:   <<< fp16_opt_level: O1
2022-06-14 15:06:42,970:INFO:   <<< freeze_layer_num: 0
2022-06-14 15:06:42,970:INFO:   <<< gradient_accumulation_steps: 1
2022-06-14 15:06:42,970:INFO:   <<< hard_negative_rate: 0.5
2022-06-14 15:06:42,970:INFO:   <<< init_model: ckpts/ckpt_msvd_retrieval_looseType/pytorch_model.bin.2
2022-06-14 15:06:42,970:INFO:   <<< linear_patch: 2d
2022-06-14 15:06:42,970:INFO:   <<< local_rank: 0
2022-06-14 15:06:42,970:INFO:   <<< loose_type: True
2022-06-14 15:06:42,970:INFO:   <<< lr: 0.0001
2022-06-14 15:06:42,970:INFO:   <<< lr_decay: 0.9
2022-06-14 15:06:42,970:INFO:   <<< margin: 0.1
2022-06-14 15:06:42,970:INFO:   <<< max_frames: 12
2022-06-14 15:06:42,971:INFO:   <<< max_words: 32
2022-06-14 15:06:42,971:INFO:   <<< n_display: 50
2022-06-14 15:06:42,971:INFO:   <<< n_gpu: 1
2022-06-14 15:06:42,971:INFO:   <<< n_pair: 1
2022-06-14 15:06:42,971:INFO:   <<< negative_weighting: 1
2022-06-14 15:06:42,971:INFO:   <<< num_thread_reader: 2
2022-06-14 15:06:42,971:INFO:   <<< output_dir: ckpts/ckpt_msvd_retrieval_looseType
2022-06-14 15:06:42,971:INFO:   <<< pretrained_clip_name: ViT-B/32
2022-06-14 15:06:42,971:INFO:   <<< rank: 0
2022-06-14 15:06:42,971:INFO:   <<< resume_model: None
2022-06-14 15:06:42,971:INFO:   <<< sampled_use_mil: False
2022-06-14 15:06:42,971:INFO:   <<< seed: 42
2022-06-14 15:06:42,971:INFO:   <<< sim_header: meanP
2022-06-14 15:06:42,972:INFO:   <<< slice_framepos: 2
2022-06-14 15:06:42,972:INFO:   <<< task_type: retrieval
2022-06-14 15:06:42,972:INFO:   <<< text_num_hidden_layers: 12
2022-06-14 15:06:42,972:INFO:   <<< train_csv: data/.train.csv
2022-06-14 15:06:42,972:INFO:   <<< train_frame_order: 0
2022-06-14 15:06:42,972:INFO:   <<< use_mil: False
2022-06-14 15:06:42,972:INFO:   <<< val_csv: data/.val.csv
2022-06-14 15:06:42,972:INFO:   <<< video_dim: 1024
2022-06-14 15:06:42,972:INFO:   <<< visual_num_hidden_layers: 12
2022-06-14 15:06:42,972:INFO:   <<< warmup_proportion: 0.1
2022-06-14 15:06:42,972:INFO:   <<< world_size: 4
2022-06-14 15:06:42,972:INFO: device: cuda:0 n_gpu: 4
2022-06-14 15:06:43,997:WARNING: Stage-One:True, Stage-Two:False
2022-06-14 15:06:43,998:WARNING: Test retrieval by loose type.
2022-06-14 15:06:43,998:WARNING: 	 embed_dim: 512
2022-06-14 15:06:43,998:WARNING: 	 image_resolution: 224
2022-06-14 15:06:43,998:WARNING: 	 vision_layers: 12
2022-06-14 15:06:43,998:WARNING: 	 vision_width: 768
2022-06-14 15:06:43,998:WARNING: 	 vision_patch_size: 32
2022-06-14 15:06:43,998:WARNING: 	 context_length: 77
2022-06-14 15:06:43,998:WARNING: 	 vocab_size: 49408
2022-06-14 15:06:43,998:WARNING: 	 transformer_width: 512
2022-06-14 15:06:43,998:WARNING: 	 transformer_heads: 8
2022-06-14 15:06:43,998:WARNING: 	 transformer_layers: 12
2022-06-14 15:06:43,998:WARNING: 		 linear_patch: 2d
2022-06-14 15:06:43,998:WARNING: 	 cut_top_layer: 0
2022-06-14 15:06:45,482:WARNING: 	 sim_header: meanP
2022-06-14 15:07:03,028:INFO: ***** Running test *****
2022-06-14 15:07:03,028:INFO:   Num examples = 27763
2022-06-14 15:07:03,028:INFO:   Batch size = 16
2022-06-14 15:07:03,028:INFO:   Num steps = 1736
2022-06-14 15:07:03,028:INFO: ***** Running val *****
2022-06-14 15:07:03,028:INFO:   **Stopping Here**..Num examples = 4290

Also, I found that my GPU usage : 3,4,5,6 is on 100% in all.  
my command : 
DATA_PATH=/home/key2317/CLIP4Clip/msvd_data
CUDA_VISIBLE_DEVICES=3,4,5,6 python -m torch.distributed.launch --nproc_per_node=4 \
main_task_retrieval.py --do_train --num_thread_reader=2 \
--epochs=5 --batch_size=128 --n_display=50 \
--data_path ${DATA_PATH} \
--features_path ${DATA_PATH}/MSVD_Videos \
--output_dir ckpts/ckpt_msvd_retrieval_looseType \
--init_model ckpts/ckpt_msvd_retrieval_looseType/pytorch_model.bin.2 \
--lr 1e-4 --max_words 32 --max_frames 12 --batch_size_val 16 \
--datatype msvd \
--feature_framerate 1 --coef_lr 1e-3 \
--freeze_layer_num 0 --slice_framepos 2 \
--loose_type --linear_patch 2d --sim_header meanP \
--pretrained_clip_name ViT-B/32

**Comment by celestialxevermore** [comment]
well, I tried a little print("logging") between 

    print("before DDP Problem")
    model = torch.nn.parallel.DistributedDataParallel(model, device_ids=[local_rank],
                                                      output_device=local_rank, find_unused_parameters=True)
    
    print("after DDP Problem")

but I cannot see that " print("after DDP Problem") " is written on my line. 

I think that there is problem on DDP. Is that any idea do you have on that?
